api/src/main/java/io/grpc/Grpc.java

```diff
@@ -62,4 +64,64 @@ private Grpc() {
   @Retention(RetentionPolicy.SOURCE)
   @Documented
   public @interface TransportAttr {}
+
+  /**
+   * Creates a channel builder with a target string and credentials. The target can be either a
+   * valid {@link NameResolver}-compliant URI, or an authority string.
+   *
+   * <p>A {@code NameResolver}-compliant URI is an absolute hierarchical URI as defined by {@link
+   * java.net.URI}. Example URIs:
+   * <ul>
+   *   <li>{@code "dns:///foo.googleapis.com:8080"}</li>
+   *   <li>{@code "dns:///foo.googleapis.com"}</li>
+   *   <li>{@code "dns:///%5B2001:db8:85a3:8d3:1319:8a2e:370:7348%5D:443"}</li>
+   *   <li>{@code "dns://8.8.8.8/foo.googleapis.com:8080"}</li>
+   *   <li>{@code "dns://8.8.8.8/foo.googleapis.com"}</li>
+   *   <li>{@code "zookeeper://zk.example.com:9900/example_service"}</li>
+   * </ul>
+   *
+   * <p>An authority string will be converted to a {@code NameResolver}-compliant URI, which has
+   * the scheme from the name resolver with the highest priority (e.g. {@code "dns"}),
+   * no authority, and the original authority string as its path after properly escaped.
+   * We recommend libraries to specify the schema explicitly if it is known, since libraries cannot
+   * know which NameResolver will be default during runtime.
+   * Example authority strings:
+   * <ul>
+   *   <li>{@code "localhost"}</li>
+   *   <li>{@code "127.0.0.1"}</li>
+   *   <li>{@code "localhost:8080"}</li>
+   *   <li>{@code "foo.googleapis.com:8080"}</li>
+   *   <li>{@code "127.0.0.1:8080"}</li>
+   *   <li>{@code "[2001:db8:85a3:8d3:1319:8a2e:370:7348]"}</li>
+   *   <li>{@code "[2001:db8:85a3:8d3:1319:8a2e:370:7348]:443"}</li>
+   * </ul>
+   */
+  @ExperimentalApi("TODO")
+  public static ManagedChannelBuilder<?> newChannelBuilder(
+      String target, ChannelCredentials creds) {
+    return ManagedChannelRegistry.getDefaultRegistry().newChannelBuilder(target, creds);
+  }
+
+  /**
+   * Creates a channel builder from a host, port, and credentials. The host and port are combined to
+   * form an authority string and then passed to {@link #newChannelBuilder(String,
+   * ChannelCredentials)}. IPv6 addresses are properly surrounded by square brackets ("[]").
+   */
+  @ExperimentalApi("TODO")
+  public static ManagedChannelBuilder<?> newChannelBuilderForAddress(
+      String host, int port, ChannelCredentials creds) {
+    return newChannelBuilder(authorityFromHostAndPort(host, port), creds);
+  }
+
+  /**
+   * Combine a host and port into an authority string.
+   */
+  // A copy of GrpcUtil.authorityFromHostAndPort
```

Comment:
Couldn't we eliminate copies? e.g. keep this and remove the one in `io.grpc.internal.GrpcUtil` .

Reply:
ok